src/main/scala/akka/persistence/cassandra/query/EventsByTagFetcher.scala

```diff
@@ -0,0 +1,131 @@
+package akka.persistence.cassandra.query
+
+import java.nio.ByteBuffer
+import java.util.UUID
+import scala.concurrent.Future
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.actor.Props
+import akka.actor.Status
+import akka.pattern.pipe
+import akka.persistence.PersistentRepr
+import akka.serialization.SerializationExtension
+import com.datastax.driver.core.PreparedStatement
+import com.datastax.driver.core.ResultSet
+import com.datastax.driver.core.Session
+import com.datastax.driver.core.utils.Bytes
+import akka.actor.ActorLogging
+import scala.annotation.tailrec
+import akka.actor.DeadLetterSuppression
+
+private[query] object EventsByTagFetcher {
+
+  private final case class InitResultSet(rs: ResultSet)
+  private case object Fetched extends DeadLetterSuppression
+
+  def props(tag: String, timeBucket: String, fromOffset: UUID, toOffset: UUID, limit: Int, replyTo: ActorRef,
+            session: Session, preparedSelect: PreparedStatement, seqNumbers: SequenceNumbers,
+            settings: CassandraReadJournalConfig): Props =
+    Props(new EventsByTagFetcher(tag, timeBucket, fromOffset, toOffset, limit, replyTo, session, preparedSelect,
+      seqNumbers, settings)).withDispatcher(settings.pluginDispatcher)
+
+}
+
+private[query] class EventsByTagFetcher(
+  tag: String, timeBucket: String, fromOffset: UUID, toOffset: UUID, limit: Int,
+  replyTo: ActorRef, session: Session, preparedSelect: PreparedStatement,
+  seqN: SequenceNumbers, settings: CassandraReadJournalConfig)
+  extends Actor with ActorLogging {
+
+  import context.dispatcher
+  import akka.persistence.cassandra.listenableFutureToFuture
+  import akka.persistence.cassandra.query.UUIDComparator.comparator.compare
+  import EventsByTagFetcher._
+  import EventsByTagPublisher._
+
+  val serialization = SerializationExtension(context.system)
+
+  def persistentFromByteBuffer(b: ByteBuffer): PersistentRepr = {
+    serialization.deserialize(Bytes.getArray(b), classOf[PersistentRepr]).get
+  }
+
+  var highestOffset: UUID = fromOffset
+  var count = 0
+  var seqNumbers = seqN
+
+  override def preStart(): Unit = {
+    val boundStmt = preparedSelect.bind(tag, timeBucket, fromOffset, toOffset, limit: Integer)
+    boundStmt.setFetchSize(2)
```

Comment:
This should probably either be the configured fetchSize or maybe based on the buffer size?